modules/orchestrator/src/main/java/org/eclipse/xpanse/orchestrator/utils/CredentialApiUtil.java

```diff
@@ -0,0 +1,318 @@
+        createCredentialApi(credentialMap, csp);
+    }
+
+    /**
+     * delete credentialApi for plugins.
+     */
+    public void deleteCredentialApi(Csp csp) {
+        File file = new File(getCredentialApiDir(), csp + "_" + "credentialApi" + ".html");
+        if (file.exists()) {
+            log.info("Delete credentialApi html file:{}, success:{}", file.getName(),
+                    file.delete());
+        }
+    }
+
+    private Map getProMap(List<CredentialVariable> credentialVariables) {
+        Map<String, Map<String, String>> credentialVariableProMap = new HashMap<>();
+        for (CredentialVariable credentialVariable : credentialVariables) {
+            Map credentialVariableMap = new HashMap();
+            credentialVariableMap.put("example", credentialVariable.getValue());
+            credentialVariableMap.put("type", "string");
+            credentialVariableMap.put("description", credentialVariable.getDescription());
+            credentialVariableProMap.put(credentialVariable.getName(), credentialVariableMap);
+        }
+        return credentialVariableProMap;
+    }
+
+    private String getApiDocsJson(Map<CredentialType, CredentialDefinition> credentialMap,
+            Csp csp) {
+        String serviceUrl = getServiceUrl();
+        String variablesStr = null;
+        String variablesProStr = null;
+
+        ObjectMapper mapper = new ObjectMapper();
+        mapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);
+
+        CredentialDefinition credentialDefinition = credentialMap.get(CredentialType.VARIABLES);
+        List<CredentialVariable> variables = credentialDefinition.getVariables();
+
+        List<String> variablesNameList = variables.stream().map(CredentialVariable::getName)
+                .collect(Collectors.toList());
+        Map variableProMap = getProMap(variables);
+        String operationName = csp + "-" + credentialDefinition.getName();
+        try {
+            variablesStr = mapper.writeValueAsString(variablesNameList);
+            variablesProStr = mapper.writeValueAsString(variableProMap);
+        } catch (JsonProcessingException e) {
+            log.error("Failed to write value as string.", e);
+        }
+        //CHECKSTYLE OFF: LineLength
+        return String.format("""
+                        openapi: 3.0.1
+                        info:
+                          title: CredentialApiUtil definition
+                          version: v1.16
+                          description: CredentialApiUtil for credentials capabilities by the plugins.
+                        servers:
+                          - url: %s
+                            description: Generated server url
+                        tags:
+                          - name: Service Vendor
+                            description: APIs to manage register services.
+                        paths:
+                          /xpanse/service/%s:
```

Comment:
Each Csp defines more than one credential type in the plugin. Below the 'paths:', there should be a list that is dynamically generated based on the credential definition list obtained by Csp from plugin.

Reply:
Thanks for your suggestion,But I think the credential types supported by the plugin are fixed, so we only need to add them here without dynamically generating them. Otherwise, when generating API documents, the format of the openAPI may cause errors due to "[]", "," and character indentation. It is not worth dealing with these format and specification issues again